app/controllers/users_controller.rb

```diff
@@ -50,4 +52,18 @@ def find_user
       redirect_to users_url, alert: "user not exists"
     end
   end
+
+  def admin_role
+    unless current_user.admin?
+      redirect_to root_url, :alert => "You are not allowed"
+    end
+  end
+
+  def correct_user
+    @user = User.find_by(id: params[:id])
+    if @user.nil?
+      redirect_to root_url, :alert => "User not exists"
```

Comment:
chỗ này viết là redirect_to root_url, :alert => "User not exists" if @user.nil? cho đồng bộ với dưới
hoặc unless @user